src/server/components/charts-engine/components/processor/control-builder.ts

```diff
@@ -0,0 +1,150 @@
+import type {ChartBuilder} from './types';
+
+type ChartBuilderArgs = {
+    config: {
+        data: {
+            shared: string;
+        };
+    };
+};
+
+export const getControlBuilder = async (args: ChartBuilderArgs): Promise<ChartBuilder> => {
+    const {config} = args;
+    let shared: ControlShared;
+
+    // Nothing happens here - just for compatibility with the editor
+    const emptyStep = (name: string) => async (options: {params: StringParams}) => {
+        const {params} = options;
+        const timeStart = process.hrtime();
+        const context = getChartApiContext({
+            name,
+            shared,
+            params,
+            actionParams: {} as Record<string, string | string[]>,
+            userLang: null,
+        });
+
+        return {
+            exports: {},
+            executionTiming: process.hrtime(timeStart),
+            name,
+            runtimeMetadata: context.__runtimeMetadata,
+        };
+    };
+
+    const ChartBuilder: ChartBuilder = {
+        buildShared: async () => {
+            if (typeof config.data.shared === 'string') {
+                shared = JSON.parse(config.data.shared);
+            } else {
+                shared = config.data.shared;
+            }
+        },
+        buildModules: async () => {
+            return {};
+        },
+        buildParams: async (options) => {
+            const {params} = options;
+            const timeStart = process.hrtime();
+            const context = getChartApiContext({
+                name: 'Params',
+                shared,
+                params,
+                actionParams: {} as Record<string, string | string[]>,
+                userLang: null,
+            });
+            return {
+                exports: {},
+                executionTiming: process.hrtime(timeStart),
+                name: 'Params',
+                runtimeMetadata: context.__runtimeMetadata,
+            };
+        },
+        buildUrls: async (options) => {
+            const {params} = options;
+            const timeStart = process.hrtime();
+
+            const context = getChartApiContext({
+                name: 'Urls',
+                shared,
+                params,
+                actionParams: {} as Record<string, string | string[]>,
+                userLang: null,
+            });
+
+            return {
+                exports: buildSources({
+                    shared,
+                    params,
+                }),
+                executionTiming: process.hrtime(timeStart),
+                name: 'Urls',
+                runtimeMetadata: context.__runtimeMetadata,
+            };
+        },
+
+        buildChartLibraryConfig: emptyStep('Highcharts'),
+
+        buildChartConfig: emptyStep('Config'),
+
+        buildChart: async (options) => {
+            const {data, params} = options;
+            const timeStart = process.hrtime();
+
+            const context = getChartApiContext({
+                name: 'JavaScript',
```

Comment:
It will be nice to create const enum for tab names. I will do it by myself later

Reply:
Discussed to fix it later at whole usages